### ISSUE TITLE
False Positive from S1172 with value tuples

### ISSUE THREAD

**Issue by shuebner** [SonarSource/sonar-csharp#1049]
### Description

S1172 gives a false positive on a method parameter which is an array of value tuples and is used in a foreach loop via deconstruction.

### Repro steps

```csharp
    public class Class1
    {
        public void Baz()
        {
            this.Foo(new[] { ("some", "thing") });
        }

        private void Foo((string key, string value)[] bars)
        {
            foreach (var (key, value) in bars) {}
        }
    }
```
### Expected behavior

S1172 does not complain about `bars` being unused.

### Actual behavior

S1172 complains about `bars` being unused.

### Related information

* SonarC# Version 6.7.1.4347
* Visual Studio Version 15.5.0

**Comment by valhristov** [comment]
Thanks for the feedback, @shuebner. This is caused by our limited support for C# 7 syntax at the moment. It is on our priority list for SonarC#, though, and we will be working on it in the upcoming months.
